Makefile

```diff
@@ -52,7 +59,12 @@ check:
 vendor: check
 	# download the dependencies
 	git submodule update --init
-	$(MAKE) -C vendor all
+	# create+activate a Python venv if not already active
+	if [ -z "$(VIRTUAL_ENV)" ]; then \
+	$(PYTHON) -m venv $(SHAREDIR)/venv && \
+	. $(SHAREDIR)/venv/bin/activate && \
+	$(SHAREDIR)/venv/bin/pip install -U pip; \
```

Comment:
```suggestion
	pip install -U pip; \
```

Reply:
No, that was intentional. It's not guaranteed that `pip` will resolve to the newly created venv version right away. If the shell supports caching/hashing, and there was some other `pip` in PATH before, then that would be used (unless inserting additional steps like `hash -r`).

Reply:
The commands in `Makefile` are executed in a new shell which does not have a cache entry for `pip`.

Reply:
You don't know what exactly gets inherited by the parent shell. Better safe than sorry.

Reply:
I prefer to stick to the usual pattern: activate a virtual environment and then run `pip` without adding the path.

Reply:
> https://github.com/UB-Mannheim/ocr-fileformat/pull/162/commits/63dbaea439d235d03d0c7bd0a267c7a3ef464a50

@stweil this is kind of hostile, you know. I had a different opinion. You just piggy-backed this and immediately merged without giving me any chance.

Reply:
@bertsky, you wanted to get this pull request merged as soon as possible. Here I only fixed the code to our standard pattern which we have anywhere else and which is completely safe, so no need for "better safe than sorry". And by the way, in my opinion setting an open discussion to "resolved" is indeed a strange action.